libs/sdk-common/src/lnurl/specs/auth.rs

```diff
@@ -6,10 +6,12 @@ use reqwest::Url;
 
 use crate::prelude::*;
 
+#[tonic::async_trait]
 pub trait LnurlAuthSigner {
-    fn derive_bip32_pub_key(&self, derivation_path: &[ChildNumber]) -> LnUrlResult<Vec<u8>>;
-    fn sign_ecdsa(&self, msg: &[u8], derivation_path: &[ChildNumber]) -> LnUrlResult<Vec<u8>>;
-    fn hmac_sha256(
+    async fn derive_bip32_pub_key(&self, derivation_path: &[ChildNumber]) -> LnUrlResult<Vec<u8>>;
+    async fn sign_ecdsa(&self, msg: &[u8], derivation_path: &[ChildNumber])
+        -> LnUrlResult<Vec<u8>>;
+    async fn hmac_sha256(
```

Comment:
If we make these async we will have to follow up with a change in breez_sdk_liquid that also uses this trait.

Reply:
This comes from the mutex for the signer. Since these functions use the signer, and the signer is behind a async mutex, this function needs to be async too.